NOX-416 Change Create and Update Dto make all Fields nullable

anna-naumova25 opened the pull request.

anna-naumova25 committed:
NOX-416 Change Create and Update Dto make all Fields nullable

anna-naumova25 committed:
Merge remote-tracking branch 'origin/main' into feature/NOX-416

gjorgji-grgovski-authoritypartners commented on src/Nox.Generator/Application/Dto/EntityCreateDto.template.cs:
```diff
@@ -42,11 +48,7 @@ public abstract class {{className}}Base : IEntityDto<DomainNamespace.{{entity.Na
-    {{- else -}}
-    public {{SinglePrimitiveTypeForKey key}} {{key.Name}} { get; set; } = default!;
-    {{- end}}
+    public {{keyPrimitiveType key}}? {{key.Name}} { get; set; }
```
```suggestion
    public virtual {{keyPrimitiveType key}}? {{key.Name}} { get; set; }
```

rochar replied:
@anna-naumova25  I have applied the suggestion, so if builds I will merge it

gjorgji-grgovski-authoritypartners reviewed (dismissed):
Looks good, just a small comment to make the key property virtual in the EntityCreateDto

rochar committed:
Update src/Nox.Generator/Application/Dto/EntityCreateDto.template.cs

Co-authored-by: Gjorgji Grgovski <131167535+gjorgji-grgovski-authoritypartners@users.noreply.github.com>

rochar committed:
remove assert file content

this tests are not bring value just a lot of work
its enough to assert the file was generated
ClientApiTests ensure the code compileas and the behavior

rochar committed:
Merge branch 'main' into feature/NOX-416
